### DISCUSSION TITLE
What makes a GREAT GitHub repo?

### DISCUSSION THREAD

**Discussion by eddiejaoude** [EddieHubCommunity/HealthCheck#12] [Ideas]
For the Open Source HealthCheck project, we will be performing checks on the Repo. What **checks** make a GREAT repo? 👇 

This is what we have implemented on the app so far...

- Is there a concise description?
- Does the repo have a url?
- Are there a good number of repo topics?
- How active is the project?
- Are there open issues?
... *what else, share your ideas below* 👇 

<img width="1269" alt="Screenshot 2024-07-10 at 14 50 04" src="https://github.com/EddieHubCommunity/HealthCheck/assets/624760/517186cb-74a5-4f3c-8b59-cd64f5f5eb0b">

**Comment by Pradumnasaraf** [comment]
Some more can be:

- Does it have a licence, Code of Conduct and Contributing guidelines?
- Does the repo version their releases or use tags 
- Does it have contributors (Can be subjective)
- Does it have any forks?

**Reply by eddiejaoude** [reply]
Great ideas, thank you 👍

**Comment by amandamartin-dev** [comment]
What about...
Number of assigned issues (or unassigned)
Age of repo (for example, I wouldn't use code from something very new sub-v1 in a production project)
Number of stale issues (subjective - how long sitting open or waiting for PR makes something stale)
out of date dependencies?

**Reply by eddiejaoude** [reply]
Love those! Thank you Amanda 🔥

**Comment by Idrinth** [comment]
- docs folder may be a bonus
- docs branch may be a bonus

**Reply by eddiejaoude** [reply]
Great idea to check both of those 👍  I much prefer having a `docs` folder to the wiki

**Comment by eddiejaoude** [comment]
Some more ideas I thought of 

- does it have a contributing guide
- does it have a repo rater entry/badge (i am biased on this one 😂 )
- are pull requests getting reviewed/merged
- when was the last release created (recently or not)

**Reply by Idrinth** [reply]
to be fair not everything needs releases, for example the docs site of IAB never has a release, since the relevant version for the docs is the version of the tooling - we could of course release every merge request merged, but that seems excessive and without much benefit to anyone.

**Reply by eddiejaoude** [reply]
yes probably not its own release. But if the docs site its own repo, I still think releases should be created for this repo which should match the release version of the tooling

**Comment by Muhammad-Owais-Warsi** [comment]
Here are few in mind -

- are the contributors following the contributors guidelines ?
- how active are the maintainers ?
- are beginners welcome or not ?  Is their any scope for learning for them or not ?

**Reply by eddiejaoude** [reply]
Thank you @Muhammad-Owais-Warsi, I can see how number 2 can be checked, but I am not sure how number 1 and 3 could be checked, could you elaborate more?

**Comment by hatemhosny** [comment]
Good README, with:
- Project description 
- Link to demo
- Screenshots
- Usage (bonus for code snippets)
- Link to docs for more details
- Guide for building/running the project

... rather than just go to docs to know what this project is about. Or even worse, nothing in README and no docs.

**Reply by eddiejaoude** [reply]
Thanks for the suggestion, can you think of a way to check these programmatically using the GitHub API?

**Reply by hatemhosny** [reply]
You can get the README file content using the API, but then you would need to parse it yourself. 
Or, if you want to get fancy, you can use an AI model to see if the content satisfies your checks. The response can be cached to avoid repeated checks if the content has not changed.

Also, JSR has recently introduced the concept of [JSR score](https://jsr.io/docs/scoring), which is automatically calculated. It is very specific to JS/TS libraries, but can give you some ideas/inspiration.
